gobblin-metastore/src/main/resources/gobblin_job_history_store.sql

```diff
@@ -91,29 +91,29 @@ CREATE TABLE IF NOT EXISTS gobblin_task_metrics (
 );
 
 CREATE TABLE IF NOT EXISTS gobblin_job_properties (
-  property_id BIGINT(21) NOT NULL AUTO_INCREMENT,
-  job_id VARCHAR(128) NOT NULL,
-  property_key VARCHAR(128) NOT NULL,
-  property_value TEXT NOT NULL,
+	property_id BIGINT(21) NOT NULL AUTO_INCREMENT,
+	job_id VARCHAR(128) NOT NULL,
+	property_key VARCHAR(128) NOT NULL,
+	property_value MEDIUMTEXT NOT NULL,
```

Comment:
Please correct me if I am wrong, this (MEDIUMTEXT column type) will add about ~16 MB per record in the table irrespective of property value size. 

However, most of the property values will be much smaller < 100 bytes, so the table will grow quite rapidly for regularly running Gobblin instance. 

And for property: source.filebased.fs.snapshot, it seems to refer to all the files listed in current snapshot (before distributing them across workunits), which can shoot beyond ~16k files limit for a big dataset (assuming 100 bytes per file path) 

However, since this fix patches an issue which would otherwise break the history store quite easily, I think we can merge this in and create another issue to track the concerns listed above. Thoughts?